pkg/zonemgr/zone_file.go

```diff
@@ -19,3 +23,26 @@ func NewZoneFile(fileName string) *ZoneFile {
 func (zoneFile *ZoneFile) writeFile(content string) (err error) {
 	return os.WriteFile(zoneFile.zoneFileFullName, []byte(content), zoneFilePerm)
 }
+
+func (zoneFile *ZoneFile) readFile() ([]byte, error) {
+	return os.ReadFile(zoneFile.zoneFileFullName)
+}
+
+func (zoneFile *ZoneFile) ReadSoaSerial() int {
+	serial := 0
+	if content, err := zoneFile.readFile(); err == nil && content != nil {
```

Comment:
We shouldn't ignore errors. If the error is that the file doesn't exist, it is ok, and we should return `nil` soaSerial. But for any other error we should return an error and not proceed as if nothing happened.

Reply:
done